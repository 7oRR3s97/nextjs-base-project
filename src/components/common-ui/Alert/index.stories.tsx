import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import { within, fireEvent } from '@storybook/testing-library';

import AlertComponent from './index';

export default {
  title: 'CommonUI/Alert',
  component: AlertComponent,
  argTypes: { onClose: { action: 'closed' } },
  decorators: [
    (Story) => (
      <div style={{ position: 'relative', width: '100%', height: '100%' }}>
        <Story />
      </div>
    ),
  ],
} as ComponentMeta<typeof AlertComponent>;

const Template: ComponentStory<typeof AlertComponent> = (args) => (
  <AlertComponent data-testid="alert-component" {...args} />
);

export const Success = Template.bind({});
Success.args = {
  severity: 'success',
  open: true,
  title: 'alert',
  message: 'alert message',
  action: false,
  progress: 5000,
  sx: { position: 'unset' },
  disablePortal: true,
};
Success.play = async ({ canvasElement }) => {
  const canvas = within(canvasElement);
  fireEvent.click(canvas.getByTitle('Close'));
};

export const Info = Template.bind({});
Info.args = {
  ...Success.args,
  severity: 'info',
};
Info.play = async ({ canvasElement }) => {
  const canvas = within(canvasElement);
  fireEvent.click(canvas.getByTitle('Close'));
};

export const Error = Template.bind({});
Error.args = {
  ...Success.args,
  severity: 'error',
};
Error.play = async ({ canvasElement }) => {
  const canvas = within(canvasElement);
  fireEvent.click(canvas.getByTitle('Close'));
};

export const Warning = Template.bind({});
Warning.args = {
  ...Success.args,
  severity: 'warning',
};
Warning.play = async ({ canvasElement }) => {
  const canvas = within(canvasElement);
  fireEvent.click(canvas.getByTitle('Close'));
};
