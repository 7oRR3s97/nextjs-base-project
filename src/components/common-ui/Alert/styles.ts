import { styled } from '@mui/material/styles';
import Dialog from '@mui/material/Dialog';

export const DialogContainer = styled(Dialog)(({ theme }) => ({
  left: 'unset',
  top: 'unset',
  [theme.breakpoints.up('md')]: {
    padding: theme.spacing(4),
  },
  '.MuiDialog-paper': {
    margin: 'unset',
    maxHeight: 'unset',
  },
}));
