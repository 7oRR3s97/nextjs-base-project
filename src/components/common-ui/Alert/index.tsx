import React from 'react';
import { AlertColor } from '@mui/material/Alert';
import AlertTitle from '@mui/material/AlertTitle';
import type { DialogProps } from '@mui/material/Dialog';

import Collapse from 'components/common-ui/Collapse';
import AlertContent from './components/AlertContent';

import { DialogContainer } from './styles';

interface Props extends DialogProps {
  open: boolean;
  severity: AlertColor;
  onClose(): void;
  title: string;
  message?: string;
  action?: boolean;
  actionComponent?: React.ReactNode;
  progress?: number;
}

const Alert: React.FC<Props> = ({
  open,
  severity,
  onClose,
  title,
  message,
  action,
  actionComponent,
  progress,
  ...props
}) => {
  const [timeOutId, setTimeOutId] = React.useState<
    NodeJS.Timeout | undefined
  >();

  const progressTimeOut = () => {
    if (!progress || !onClose) return;
    return setTimeOutId(
      setTimeout(() => {
        onClose();
      }, progress)
    );
  };

  const handleClose = () => {
    if (timeOutId) {
      clearTimeout(timeOutId);
    }
    onClose();
  };

  return (
    <DialogContainer
      {...props}
      open={open}
      TransitionComponent={Collapse}
      TransitionProps={{
        in: open,
        onEntered: progressTimeOut,
        timeout: 300,
      }}
      onClose={handleClose}
      hideBackdrop
      disableAutoFocus
      disableEnforceFocus
    >
      <AlertContent
        severity={severity}
        onClose={handleClose}
        action={action}
        actionComponent={actionComponent}
      >
        <AlertTitle>{title}</AlertTitle>
        {message}
      </AlertContent>
    </DialogContainer>
  );
};

export default Alert;
