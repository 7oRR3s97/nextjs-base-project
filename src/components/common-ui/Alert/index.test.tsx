import React from 'react';
import {
  render,
  screen,
  cleanup,
  fireEvent,
  act,
  advanceTimersByTime,
} from 'helpers/test-utils';
import { axe, toHaveNoViolations } from 'jest-axe';
import { composeStories } from '@storybook/testing-react';
import Alert from './index';

import * as AlertStories from './index.stories';

expect.extend(toHaveNoViolations);

describe('Alert', () => {
  let open = true;
  const handleOpen = (value: boolean) => {
    open = value;
  };

  beforeEach(() => {
    open = true;
  });

  afterEach(() => {
    cleanup();
    jest.clearAllTimers();
  });

  const stories = Object.values(composeStories(AlertStories)).map((Story) => {
    return {
      name: Story.storyName!,
      Story,
    };
  });

  it.each(stories)(
    'should have no accessibility error in $name alert story',
    async ({ Story }) => {
      render(<Story data-testid="alert-component" />);
      const alert = screen.getByTestId('alert-component');

      const AxeResults = await axe(alert);
      expect(AxeResults).toHaveNoViolations();
      expect(alert).toMatchSnapshot();
    }
  );

  it.each(stories)(
    'should run $name alert story play function and set open to false',
    async ({ Story }) => {
      jest.useFakeTimers();
      render(
        <Story
          data-testid="alert-component"
          onClose={() => handleOpen(false)}
        />
      );

      await advanceTimersByTime(500);

      await act(async () => {
        await Story.play({
          canvasElement: screen.getByTestId('alert-component'),
        });
      });

      expect(open).toBeFalsy();
    }
  );

  it("should render an alert and then automatically closes it using the component's progressTimeOut function", async () => {
    jest.useFakeTimers();
    render(
      <Alert
        data-testid="alert-component"
        open={open}
        onClose={() => handleOpen(false)}
        severity="error"
        title="error-title"
        message="error-message"
        progress={500}
      />
    );

    const alert = screen.queryByTestId('alert-component');
    expect(alert).toBeInTheDocument();
    expect(alert).toMatchSnapshot();

    await advanceTimersByTime(500);
    expect(open).toBeFalsy();
  });

  it("should not render the alert and then render it without progress, so the component's progressTimeOut function returns undefined", async () => {
    jest.useFakeTimers();
    open = false;

    const { rerender } = render(
      <Alert
        data-testid="alert-component"
        open={open}
        onClose={() => handleOpen(false)}
        severity="error"
        title="error-title"
        message="error-message"
      />
    );

    const nonExistingAlert = screen.queryByTestId('alert-component');

    expect(nonExistingAlert).toMatchSnapshot();
    expect(nonExistingAlert).not.toBeInTheDocument();

    handleOpen(true);
    expect(open).toBeTruthy();

    rerender(
      <Alert
        data-testid="alert-component"
        open={open}
        onClose={() => handleOpen(false)}
        severity="error"
        title="error-title"
        message="error-message"
      />
    );

    await advanceTimersByTime(500);

    expect(open).toBeTruthy();

    const existingAlert = screen.queryByTestId('alert-component');
    expect(existingAlert).toMatchSnapshot();
    expect(existingAlert).toBeInTheDocument();
  });

  it('should close the alert before the progressTimeOut function finishes, triggering a clearTimeOut execution', async () => {
    jest.useFakeTimers();
    render(
      <Alert
        data-testid="alert-component"
        open={open}
        onClose={() => handleOpen(false)}
        severity="error"
        title="error-title"
        message="error-message"
        progress={1000}
      />
    );

    await advanceTimersByTime(500);

    fireEvent.click(screen.getByTitle('Close'));
    expect(open).toBeFalsy();
  });

  it('should render an alert with the action component', () => {
    render(
      <Alert
        data-testid="alert-component"
        open={open}
        onClose={() => handleOpen(false)}
        severity="error"
        title="error-title"
        message="error-message"
        action
      />
    );

    const alert = screen.queryByTestId('alert-component');
    expect(alert).toMatchSnapshot();
    expect(alert).toBeInTheDocument();
  });
});
