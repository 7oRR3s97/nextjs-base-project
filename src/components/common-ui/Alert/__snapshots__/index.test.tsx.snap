// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert should have no accessibility error in Error alert story 1`] = `
<div
  class="MuiModal-root MuiDialog-root css-19aa5gy-MuiModal-root-MuiDialog-root"
  data-testid="alert-component"
  role="presentation"
>
  <div
    data-test="sentinelStart"
    tabindex="0"
  />
  <div
    class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-pwcg7p-MuiCollapse-root"
    role="presentation"
    style="min-height: 0px; height: auto;"
    tabindex="-1"
  >
    <div
      class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
    >
      <div
        class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
      >
        <div
          class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
        >
          <div
            aria-labelledby=":r2:"
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm css-lzee2o-MuiPaper-root-MuiDialog-paper"
            role="dialog"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardError MuiAlert-standard css-16910ee-MuiPaper-root-MuiAlert-root"
              role="alert"
            >
              <div
                class="MuiAlert-icon css-9a6hnt-MuiAlert-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="ErrorOutlineIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
              </div>
              <div
                class="MuiAlert-message css-19yrdgc-MuiAlert-message"
              >
                <div
                  class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom MuiAlertTitle-root css-qh0wya-MuiTypography-root-MuiAlertTitle-root"
                >
                  alert
                </div>
                alert message
              </div>
              <div
                class="MuiAlert-action css-ki1hdl-MuiAlert-action"
              >
                <button
                  aria-label="Close"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeSmall css-51dspl-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  title="Close"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                    data-testid="CloseIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    data-test="sentinelEnd"
    tabindex="0"
  />
</div>
`;

exports[`Alert should have no accessibility error in Info alert story 1`] = `
<div
  class="MuiModal-root MuiDialog-root css-19aa5gy-MuiModal-root-MuiDialog-root"
  data-testid="alert-component"
  role="presentation"
>
  <div
    data-test="sentinelStart"
    tabindex="0"
  />
  <div
    class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-pwcg7p-MuiCollapse-root"
    role="presentation"
    style="min-height: 0px; height: auto;"
    tabindex="-1"
  >
    <div
      class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
    >
      <div
        class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
      >
        <div
          class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
        >
          <div
            aria-labelledby=":r1:"
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm css-lzee2o-MuiPaper-root-MuiDialog-paper"
            role="dialog"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardInfo MuiAlert-standard css-1p38uc8-MuiPaper-root-MuiAlert-root"
              role="alert"
            >
              <div
                class="MuiAlert-icon css-9a6hnt-MuiAlert-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="InfoOutlinedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20, 12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10, 10 0 0,0 12,2M11,17H13V11H11V17Z"
                  />
                </svg>
              </div>
              <div
                class="MuiAlert-message css-19yrdgc-MuiAlert-message"
              >
                <div
                  class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom MuiAlertTitle-root css-qh0wya-MuiTypography-root-MuiAlertTitle-root"
                >
                  alert
                </div>
                alert message
              </div>
              <div
                class="MuiAlert-action css-ki1hdl-MuiAlert-action"
              >
                <button
                  aria-label="Close"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeSmall css-51dspl-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  title="Close"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                    data-testid="CloseIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    data-test="sentinelEnd"
    tabindex="0"
  />
</div>
`;

exports[`Alert should have no accessibility error in Success alert story 1`] = `
<div
  class="MuiModal-root MuiDialog-root css-19aa5gy-MuiModal-root-MuiDialog-root"
  data-testid="alert-component"
  role="presentation"
>
  <div
    data-test="sentinelStart"
    tabindex="0"
  />
  <div
    class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-pwcg7p-MuiCollapse-root"
    role="presentation"
    style="min-height: 0px; height: auto;"
    tabindex="-1"
  >
    <div
      class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
    >
      <div
        class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
      >
        <div
          class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
        >
          <div
            aria-labelledby=":r0:"
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm css-lzee2o-MuiPaper-root-MuiDialog-paper"
            role="dialog"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardSuccess MuiAlert-standard css-zikowp-MuiPaper-root-MuiAlert-root"
              role="alert"
            >
              <div
                class="MuiAlert-icon css-9a6hnt-MuiAlert-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="SuccessOutlinedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20,12A8,8 0 0,1 12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4C12.76,4 13.5,4.11 14.2, 4.31L15.77,2.74C14.61,2.26 13.34,2 12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0, 0 22,12M7.91,10.08L6.5,11.5L11,16L21,6L19.59,4.58L11,13.17L7.91,10.08Z"
                  />
                </svg>
              </div>
              <div
                class="MuiAlert-message css-19yrdgc-MuiAlert-message"
              >
                <div
                  class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom MuiAlertTitle-root css-qh0wya-MuiTypography-root-MuiAlertTitle-root"
                >
                  alert
                </div>
                alert message
              </div>
              <div
                class="MuiAlert-action css-ki1hdl-MuiAlert-action"
              >
                <button
                  aria-label="Close"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeSmall css-51dspl-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  title="Close"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                    data-testid="CloseIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    data-test="sentinelEnd"
    tabindex="0"
  />
</div>
`;

exports[`Alert should have no accessibility error in Warning alert story 1`] = `
<div
  class="MuiModal-root MuiDialog-root css-19aa5gy-MuiModal-root-MuiDialog-root"
  data-testid="alert-component"
  role="presentation"
>
  <div
    data-test="sentinelStart"
    tabindex="0"
  />
  <div
    class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-pwcg7p-MuiCollapse-root"
    role="presentation"
    style="min-height: 0px; height: auto;"
    tabindex="-1"
  >
    <div
      class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
    >
      <div
        class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
      >
        <div
          class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
        >
          <div
            aria-labelledby=":r3:"
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm css-lzee2o-MuiPaper-root-MuiDialog-paper"
            role="dialog"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardWarning MuiAlert-standard css-105rnht-MuiPaper-root-MuiAlert-root"
              role="alert"
            >
              <div
                class="MuiAlert-icon css-9a6hnt-MuiAlert-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="ReportProblemOutlinedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 5.99L19.53 19H4.47L12 5.99M12 2L1 21h22L12 2zm1 14h-2v2h2v-2zm0-6h-2v4h2v-4z"
                  />
                </svg>
              </div>
              <div
                class="MuiAlert-message css-19yrdgc-MuiAlert-message"
              >
                <div
                  class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom MuiAlertTitle-root css-qh0wya-MuiTypography-root-MuiAlertTitle-root"
                >
                  alert
                </div>
                alert message
              </div>
              <div
                class="MuiAlert-action css-ki1hdl-MuiAlert-action"
              >
                <button
                  aria-label="Close"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeSmall css-51dspl-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  title="Close"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                    data-testid="CloseIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    data-test="sentinelEnd"
    tabindex="0"
  />
</div>
`;

exports[`Alert should not render the alert and then render it without progress, so the component's progressTimeOut function returns undefined 1`] = `null`;

exports[`Alert should not render the alert and then render it without progress, so the component's progressTimeOut function returns undefined 2`] = `
<div
  class="MuiModal-root MuiDialog-root css-2fsdkr-MuiModal-root-MuiDialog-root"
  data-testid="alert-component"
  role="presentation"
>
  <div
    data-test="sentinelStart"
    tabindex="0"
  />
  <div
    class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-pwcg7p-MuiCollapse-root"
    role="presentation"
    style="min-height: 0px; height: auto;"
    tabindex="-1"
  >
    <div
      class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
    >
      <div
        class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
      >
        <div
          class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
        >
          <div
            aria-labelledby=":r9:"
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm css-lzee2o-MuiPaper-root-MuiDialog-paper"
            role="dialog"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardError MuiAlert-standard css-16910ee-MuiPaper-root-MuiAlert-root"
              role="alert"
            >
              <div
                class="MuiAlert-icon css-9a6hnt-MuiAlert-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="ErrorOutlineIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
              </div>
              <div
                class="MuiAlert-message css-19yrdgc-MuiAlert-message"
              >
                <div
                  class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom MuiAlertTitle-root css-qh0wya-MuiTypography-root-MuiAlertTitle-root"
                >
                  error-title
                </div>
                error-message
              </div>
              <div
                class="MuiAlert-action css-ki1hdl-MuiAlert-action"
              >
                <button
                  aria-label="Close"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeSmall css-51dspl-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  title="Close"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                    data-testid="CloseIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    data-test="sentinelEnd"
    tabindex="0"
  />
</div>
`;

exports[`Alert should render an alert and then automatically closes it using the component's progressTimeOut function 1`] = `
<div
  class="MuiModal-root MuiDialog-root css-2fsdkr-MuiModal-root-MuiDialog-root"
  data-testid="alert-component"
  role="presentation"
>
  <div
    data-test="sentinelStart"
    tabindex="0"
  />
  <div
    class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-pwcg7p-MuiCollapse-root"
    role="presentation"
    style="min-height: 0px; height: auto;"
    tabindex="-1"
  >
    <div
      class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
    >
      <div
        class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
      >
        <div
          class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
        >
          <div
            aria-labelledby=":r8:"
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm css-lzee2o-MuiPaper-root-MuiDialog-paper"
            role="dialog"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardError MuiAlert-standard css-16910ee-MuiPaper-root-MuiAlert-root"
              role="alert"
            >
              <div
                class="MuiAlert-icon css-9a6hnt-MuiAlert-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="ErrorOutlineIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
              </div>
              <div
                class="MuiAlert-message css-19yrdgc-MuiAlert-message"
              >
                <div
                  class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom MuiAlertTitle-root css-qh0wya-MuiTypography-root-MuiAlertTitle-root"
                >
                  error-title
                </div>
                error-message
              </div>
              <div
                class="MuiAlert-action css-ki1hdl-MuiAlert-action"
              >
                <button
                  aria-label="Close"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeSmall css-51dspl-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  title="Close"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                    data-testid="CloseIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    data-test="sentinelEnd"
    tabindex="0"
  />
</div>
`;

exports[`Alert should render an alert with the action component 1`] = `
<div
  class="MuiModal-root MuiDialog-root css-2fsdkr-MuiModal-root-MuiDialog-root"
  data-testid="alert-component"
  role="presentation"
>
  <div
    data-test="sentinelStart"
    tabindex="0"
  />
  <div
    class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-pwcg7p-MuiCollapse-root"
    role="presentation"
    style="min-height: 0px; height: auto;"
    tabindex="-1"
  >
    <div
      class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
    >
      <div
        class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
      >
        <div
          class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
        >
          <div
            aria-labelledby=":rb:"
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm css-lzee2o-MuiPaper-root-MuiDialog-paper"
            role="dialog"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardError MuiAlert-standard css-6mmzq3-MuiPaper-root-MuiAlert-root"
              role="alert"
            >
              <div
                class="MuiAlert-icon css-9a6hnt-MuiAlert-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="ErrorOutlineIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
              </div>
              <div
                class="MuiAlert-message css-19yrdgc-MuiAlert-message"
              >
                <div
                  class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom MuiAlertTitle-root css-qh0wya-MuiTypography-root-MuiAlertTitle-root"
                >
                  error-title
                </div>
                error-message
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    data-test="sentinelEnd"
    tabindex="0"
  />
</div>
`;
